src/NumericalAlgorithms/FiniteDifference/Wcns5z.cpp

```diff
@@ -12,43 +12,91 @@
 #include "Domain/Structure/Direction.hpp"
 #include "Domain/Structure/DirectionMap.hpp"
 #include "Domain/Structure/Side.hpp"
+#include "NumericalAlgorithms/FiniteDifference/FallbackReconstructorType.hpp"
+#include "NumericalAlgorithms/FiniteDifference/Minmod.hpp"
+#include "NumericalAlgorithms/FiniteDifference/MonotonisedCentral.hpp"
 #include "NumericalAlgorithms/FiniteDifference/Reconstruct.tpp"
 #include "Utilities/ErrorHandling/Error.hpp"
 #include "Utilities/GenerateInstantiations.hpp"
 #include "Utilities/Gsl.hpp"
 
 namespace fd::reconstruction {
 
+namespace {
+// use type alias for the function pointer return type for brevity
+template <size_t Dim>
+using pointer_return_type =
+    std::tuple<void (*)(gsl::not_null<std::array<gsl::span<double>, Dim>*>,
+                        gsl::not_null<std::array<gsl::span<double>, Dim>*>,
+                        const gsl::span<const double>&,
+                        const DirectionMap<Dim, gsl::span<const double>>&,
+                        const Index<Dim>&, size_t, double, size_t),
+               void (*)(gsl::not_null<DataVector*>, const DataVector&,
+                        const DataVector&, const Index<Dim>&, const Index<Dim>&,
+                        const Direction<Dim>&, const double&, const size_t&),
+               void (*)(gsl::not_null<DataVector*>, const DataVector&,
+                        const DataVector&, const Index<Dim>&, const Index<Dim>&,
+                        const Direction<Dim>&, const double&, const size_t&)>;
+
+// returns functions pointer
+template <size_t NonlinearWeightExponent, typename Reconstructor, size_t Dim>
+pointer_return_type<Dim> function_pointer_v() {
```

Comment:
why is there a `_v` suffix? This isn't a constexpr variable, which is where `_v` is normally used

Reply:
Good to know. I'll remove this suffix